grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx

```diff
@@ -62,17 +68,22 @@ const RemoteSelect = inject('store')(
 
     const [options, setOptions] = useReducer(mergeOptions, []);
 
-    useEffect(() => {
-      makeRequest(href, {}).then((data) => {
-        setOptions(getOptions(data.results || data));
-      });
+    const loadOptionsCallback = useCallback((query?: string): void => {
+      (async () => {
```

Comment:
I understand the pattern you applied but why don't you just set the inner function of `useCallback` to be async instead of all the unnecessary lines? Unless I missed something :thinking:

Reply:
good point 👍 I initially tried this but it was complaining about the return type. Addressed this [here](https://github.com/grafana/oncall/pull/1185/commits/0f5f9ebfcfb0469aafbcfc470597f07fe5d32b33)